css-naming-convention.md

```diff
@@ -0,0 +1,195 @@
+
+
+### Component
+
+github-`CommitBox`
+
+Components are building blocks of a package. They can be small or large. Components can also contain other components. It's more about seeing what belongs together.
+
+Components can also share the same elements. A pattern often found is that a new Component starts where a childElement ends. 
+
+```html
+<ul class="github-List">
+  <li class="github-List-item github-Commit">
+    <label class="github-Commit-message"></label>
+    <span class="github-Commit-time"></span>
+  </li>
+</ul>
+```
+
+In this example, `github-List-item` is responsible for the "container" and layout styles. `github-Commit` is responsible for the "content" inside.
+
+> Note: Components use PascalCase. This makes it easy to spot them in the markup. For example in `settings-List-item`, it's easy to see that the component is `List` and not `settings` or `item`.
+
+
+### Child element
+
+github-CommitBox-`commitButton`
+
+Elements that are part of a component are appended to the component's class name with a single dash (`-`).
+
+> Note: If multiple words are needed, camelCase is used: github-CommitBox`-commitButton`.
+
+
+### Modifier
+
+github-CommitBox-commitButton`--primary`
+
+Modifiers are used if a component is very similar to the default component but varies slightly. Like has a different color. Modifiers use a double dash `--` to distinguish them from components and child elements.
+
+> Note: If multiple words are needed, camelCase is used: github-CommitBox-commitButton`--primaryColor`.
+
+
+### States
+
+github-CommitBox-commitButton `is-disabled`
+
+States are prefixed with a short verb, like `is-` or `has-`. Since these class names probably get used in many different places, it should never be styled stand-alone and always be a chained selector.
+
+```less
+.is-disabled {
+  // nope
+}
+.github-CommitBox-commitButton.is-disabled {
+  // yep
+}
+```
+
+> Note: If multiple words are needed, camelCase is used: `has-collapsedItems`.
+
+
+
+## More guidelines, not crucial, but still good to know and follow. ;-)
+
+- Styling elements (like `div`) should be avoided. This makes it easier to switch elements, like from a `<button>` to `<a>`.
+- No utility classes. Themes and user styles can only override CSS but not change the markup. Therefore having utility classes doesn't make as much sense once you override them.
+- Avoid using existing CSS classes to reference elements in the DOM. That way when the markup changes, functionality and specs will less likely break. Instead use `ref` attributes (`ref="commitButton"`).
+- Avoid changing modifiers at runtime. If you're in need, consider changing the modifier class into a state. For example if you often want to switch a default button to a primary button, change the `github-CommitBox-commitButton--primary` modifier class into a `is-primary` state instead. Since state classes are decoupled from components, it's easier to change component names, but keep the states the same.
```

Comment:
I _think_ "component names" and "states" got switched in the last sentence here? I'm mostly just guessing here though as I'm very _un_ familiar with CSS.

Reply:
Oh, I see. It's actually easier to change both.

- Change component name: `github-CommitBox-commitButton is-disabled` -> `github-Button is-disabled`
- Change state name: `github-CommitBox-commitButton is-disabled` -> `github-CommitBox-commitButton is-inActive`

I'll rewrite it to mention both. Thanks. 👍